### DISCUSSION TITLE
Phase 4 - UI

### DISCUSSION THREAD

**Discussion by alvarix** [alvarix/FLogger#110] [Ideas]
@chadcrume Let's brainstorm 2-3 different layout options

**Comment by alvarix** [comment]
- [x] Clean Up current UI
- [ ] Dark mode
- [ ] Decide whether these are different skins on same code-base or forks
- [ ] Add Flogs to repo:
  - [ ] README
  - [ ] Project Blurb?

### Views
1. HP (connect)
2. Flogs
3. Flog

### ui-1: minimal
- HP: logo and connect to DBX button
- Flogs: Omnibox and file list
- Flog:
  - delete all borders
  - minimal text (prompts as icon tool tips?)
  - remove add entry button (return saves)
  - put copy and delete buttons in fly out

### ui-2: table
- HP: Lots of data on project, usage, authors like a README
- Flogs: Left side omnibox, Right side Flogs 
  - preference to switch sides?
  - Last Modified date, characters
- Flog: 
  - rule seperating entries
  - date, body, tools
  - dropdown in nav of omnibox and next 10 flogs


```

   ___  __                             
  / __\/ /  ___   __ _  __ _  ___ _ __ 
 / _\ / /  / _ \ / _` |/ _` |/ _ \ '__|
/ /  / /__| (_) | (_| | (_| |  __/ |   
\/   \____/\___/ \__, |\__, |\___|_|   
                 |___/ |___/           

                                
 _____ __                        
|   __|  |   ___ ___ ___ ___ ___ 
|   __|  |__| . | . | . | -_|  _|
|__|  |_____|___|_  |_  |___|_|  
                |___|___|        

░▒▓████████▓▒░▒▓█▓▒░      ░▒▓██████▓▒░ ░▒▓██████▓▒░ ░▒▓██████▓▒░░▒▓████████▓▒░▒▓███████▓▒░  
░▒▓█▓▒░      ░▒▓█▓▒░     ░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░      ░▒▓█▓▒░░▒▓█▓▒░ 
░▒▓█▓▒░      ░▒▓█▓▒░     ░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░      ░▒▓█▓▒░      ░▒▓█▓▒░      ░▒▓█▓▒░░▒▓█▓▒░ 
░▒▓██████▓▒░ ░▒▓█▓▒░     ░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒▒▓███▓▒░▒▓█▓▒▒▓███▓▒░▒▓██████▓▒░ ░▒▓███████▓▒░  
░▒▓█▓▒░      ░▒▓█▓▒░     ░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░      ░▒▓█▓▒░░▒▓█▓▒░ 
░▒▓█▓▒░      ░▒▓█▓▒░     ░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░      ░▒▓█▓▒░░▒▓█▓▒░ 
░▒▓█▓▒░      ░▒▓████████▓▒░▒▓██████▓▒░ ░▒▓██████▓▒░ ░▒▓██████▓▒░░▒▓████████▓▒░▒▓█▓▒░░▒▓█▓▒░ 
                                                                                            
                                                                                            

```

**Comment by alvarix** [comment]
### Alvar Tasks
- [x] remove repopack from repo
- [x] clear console warns (index prop)
- [x] Seems like accountInfo() ref used to display acocunt owner on dropboxflogs component is breaking the connection
- [ ] clean up UX
  - [x] autofocus
  - [x] body class
  - [x] hide autoc on mouseout and esc 
  - [x] alert feedback after copy
  - [ ] 

- [x] try to get account owner working
- [x] dark/light mode (set up for prefs  )
  - [x] toggle
- [x]  formatting error handlers? (bad dates)
  - [x] App should accept more date formats
- [ ] move legacy issues to discussion
- [x] state classes on top level div 
- [x] accept abbr date
- [x] App icon 
- [ ]Consult chad
  - [x] on create new file add to top of list, open it
  - [x] delete Flog
  - [ ] option to open plain text if parser fails?
  - [x] changing date regex (verified with Gemini) didnt work, error handling?

### Chad
- [x] pretext (non-date initial chars)
- [x] files in repo - readme (with project blurb), and release notes
- [x] DBX auth as popup to fix back button
- [x] remove obsolete json, etc files 
- [ ] formatting error handlers? (bad dates)
- [x] delete old branches?
- [ ] Look for options for in-code documentation for Vue components, composables, and regular JS modules and functions.
(NEW)
- [ ] persist disconnect button onto open flogs page 
- [ ] Spinner while loading entries

**Reply by chadcrume** [reply]
files in repo --> readme, blurb only — in Flog format, read only if I want to go through the trouble right now

**Reply by chadcrume** [reply]
pretext — update to extract pretext as property of Flog. Could display on the open flog

**Reply by alvarix** [reply]
@chadcrume 
Nicccccce. Cant wait to checkout it out (pun intended)

Remembered this morning how I thought it'd be cool to add some sort of flogger roadmap file to the repo also (future ideas and/or current PM lists).
WDYT

**Comment by alvarix** [comment]
Discussion:
- removing tailwind exposed error messages
- personally enjoying tailwind on my portfolio dev
  - if there is already a build process it seems to me a nice utility since it wont add much overhead

CSS logically placed  (look into vite rollup)

- [x] install cdn tailwind
- [x] light/dark mode based on sys pref
- [x] Flogger ascii logo
- [x] standardize UI (buttons,)
- [ ] organize CSS

---

### UI

- [ ]  HP
- description (...this will create flogger data folder)
- dbx button
- right sidebar project info section 

- [ ] Index
- Logout button
- omnibox
- list (remove or treat extensions)
-

**Comment by alvarix** [comment]
My Current Branched:

- [alvar/ux__add-open](https://github.com/alvarix/FLogger/tree/alvar/ux__add-open)
- [alvar/ux__open-flog](https://github.com/alvarix/FLogger/tree/alvar/ux__open-flog)
- [alvar/UI_cleanup](https://github.com/alvarix/FLogger/tree/alvar/UI_cleanup)
- [alvar/more-date-formats](https://github.com/alvarix/FLogger/tree/alvar/more-date-formats)
- light/dark

**Comment by chadcrume** [comment]
Meeting notes:

- Look into localization approach, and use it for default UI text like intro
- Delete flog
- From last time, we decided Chad would split list of flogs into template flogs and user flogs, and make template flogs read only
- Convert alerts to CSS style windows
- New flog should open the flog
- Sort list by newest to oldest by default
- Add ability to change sort, including alpha or date created/modified, and reversible
- Display pretext. This will help when date format or something else is messing up the regex split
- Incorporate new regex for date splitting

**Comment by alvarix** [comment]
Alvar cleared list, adding new ideas:

- [ ] Logo reloads Intro template with option to return to file list
- [ ] back button - avoid disconnect

**Comment by alvarix** [comment]
# Final tasks in Phase 4

## Features
- [ ] autosave
- [ ] state
     - [x] refresh looses dbx
     - [ ] back button
- [x] reveal pretext 
  - [ ] formatting 
    - [x] remove copy and add entry
    - [ ] add readonly flag


## Refactor
  - [ ]  move all date functions to composable? (entry.vue)
    - [ ] use date fns lib?
  - [ ] break complex files down into smaller units
  - [ ] documentation
    - [ ] inline documentation
    - [ ] flow graphs [mermaid](https://mermaid.js.org/)
      - [ ] saveEntry $a
  - [ ] possible to make addEntry and editEntry into one component?
## Repo Files
  - [ ]  content:
    - [ ] readme flog
    - [ ] homepage
    - [ ] github readme
  - [ ] PM flogs (dbx = flogger)
    - [ ] wishlist
    - [ ] tasklist
  - [x] readme

## Bugs
  - [ ] dark mode refresh
  - [x] parsing error - see bug convo
    - [ ] error handling?
    - [ ] show plain text
  - [ ] pop up on mobile 
  - [x] vercel dbx url

**Comment by alvarix** [comment]
Phase 5
- Refactor
- Markdown
- Routing
- Rebasing
- Tests
- Read only file UI

**Comment by chadcrume** [comment]
## New List

### Done

- [x] Reveal pretext
- [x] Remove copy and add entry

### Definitely Phase 4

- [ ] Alvar: auto resize text editing not working on mobile — 
   - [ ] Add entry textarea needs to autoresize
   - [ ] Click into entry to edit, textarea needs to autoresize (works when you copy entry)
   - [ ] Pretext textarea isn't autoresizing
- [ ] Alvar: Format pretext & Read-only file UI 
- [ ] Alvar: Dark mode refresh
---
Either:
- [ ] pop up on mobile
- [ ] Decide on and Populate repo flogs
  - [ ] About Flogger
  - [ ] Changelog? (features in alpha)
  - [x] Dev notes/wiki? (move away from GH)
    - [x] vercel URLS
---
- [ ] Chad: Refresh keeps dbx connection
- [ ] Chad: Read-only flag for repo files
- [ ] Chad: Edit pretext (rather than add read-only flag for pretext)

 


### Refactor items

- [ ] move all date functions to composable? (entry.vue)
   - [ ]  use date fns lib?
- [ ]  break complex files down into smaller units
- [ ]  documentation
   - [ ]  inline documentation
   - [ ]  flow graphs [mermaid](https://mermaid.js.org/)
      - [ ]  saveEntry $a
- [ ]  possible to make addEntry and editEntry into one component?

### Possibly roll to Phase 5

- [ ] Back button 
- [ ] Show plain text on flog parsing error
- [ ] Markdown
- [ ] Routing
- [ ] Visual feedback for copy

### Observations from Obsidian
- Templates with variables
- Merge to file feature

**Reply by alvarix** [reply]
for _auto resize text editing not working on mobile_
to avoid having to merge every commit into main just to test I have to get a vercel branch url working but not having luck. 
@chadcrume can we pair on this in next call?
Or if you can get the branch working it's
https://flogger-git-alvar-auto-resize-textareas-alvarixs-projects.vercel.app/
documenting procedure on [wiki](https://github.com/alvarix/FLogger/wiki/Authorizing-a-different-URL)

Uncaught Error: You must use PKCE when generating the authorization URL to not include a client secret
    at e.value (useDropboxFlogs-2a36fad4.js:2:46213)
    at oc (useDropboxFlogs-2a36fad4.js:2:51750)
    at useDropboxFlogs-2a36fad4.js:7:1041
